### ISSUE TITLE
experiment(agent): evaluate compact decision-oriented state abstraction for tabular Task 2

### ISSUE THREAD

**Issue by LiliWestermann** [1BlauNitrox/mle-final-project#128]
## Research question

Does a compact, decision-oriented state representation improve state reuse and
hidden-coin collection for the tabular Task 2 agent?

## Hypothesis

Replacing the current state representation with a compact representation of
danger, objectives, bomb utility and locally available movement will reduce
Q-table sparsity and improve classic coin collection without increasing
self-kills.

## Reference baseline

The completed Issue #102 Task 2 baseline remains a historical reference. Its
trained Q-table is not reused because its Task-1-derived initialization is not
compatible with the compact five-feature representation.

## Candidate

Use the following compact state:

```python
state = (
    danger_level,
    safe_directions_mask,
    coin_direction,
    crate_direction,
    bomb_status,
)
```

### Feature definitions

- danger_level:
  - `0`: the current tile is not covered by a predicted blast;
  - `1`: the tile is covered by a future blast, but the earliest lethal
    interval starts more than two actions later;
  - `2`: the tile is currently lethal or its earliest lethal interval starts
    within the next two actions.

- safe_directions_mask:
  - four bits in `UP`, `RIGHT`, `DOWN`, `LEFT` order;
  - a bit is set only if the first movement is currently traversable and at
    least one complete time-aware route from that movement reaches a tile
    outside all predicted blast intervals;
  - the mask is an input feature only and does not filter action selection or
    Bellman targets.

- coin_direction:
    - erster möglicher Schritt auf einem kürzesten Weg zur nächsten Münze

- crate_direction:
    - erster möglicher Schritt zu einem Feld, von dem eine Kiste
       bombardiert werden kann

- bomb_status:
    - 0 = keine Bombe verfügbar
    - 1 = verfügbar, aber keine Kiste erreichbar
    - 2 = nützlich, aber kein sicherer Fluchtweg
    - 3 = nützlich und sicherer Fluchtweg

## Independent variable

- Control: current 17-feature Task 2 state representation with an initially
  empty Q-table.
- Candidate: compact five-feature state representation with an initially empty
  Q-table.

Both treatments use identical zero initialization. No parent Q-values or
existing Task 2 Q-tables are reused.

## Controlled variables

Keep unchanged:

- Q-learning update rule;
- rewards;
- action masking mode;
- learning rate and discount factor;
- epsilon schedule;
- zero initialization of all Q-values;
- curriculum and episode budget;
- training and evaluation seeds;
- scenarios and opponents;
- checkpoint-selection rule;
- evaluation suites.

Both treatments must train fresh Q-tables. Existing Q-values must not be
converted or reused across state representations.

## Evaluation

Use five replicas per treatment and paired development seeds for:

- `classic`;
- `loot-crate`;
- `coin-heaven` Task 1 regression;
- deterministic repeats.

Confirmation seeds remain unused during development.

## Primary metric

Candidate-minus-control coin collection fraction on `classic`.

## Learning-efficiency diagnostics

Record for each replica:

- number of distinct Q-table states;
- total state visits;
- mean visits per state;
- fraction of states visited only once;
- fraction of evaluation decisions using previously unseen states;
- Q-table artifact size.

## Success criteria

The candidate passes if:

- classic collection mean difference is above zero;
- paired 95% CI lower bound is above zero;
- at least four of five replicas improve;
- unseen-state rate during evaluation is lower than control;
- mean visits per stored state increase;
- classic self-kill rate does not exceed control by more than `0.02`;
- Task 1 retention CI lower bound remains above `-0.05`;
- deterministic-repeat and runtime requirements pass.

## Compute budget

- 5 replicas per treatment;
- 10,000 training episodes per replica;
- 40 development seeds per scenario;
- one deterministic repeat;
- maximum 100,000 training episodes;
- maximum 2,480 evaluation episodes;
- maximum 36 CPU-hours;
- no paid resources.

## Out of scope

- reward changes;
- legal-action or danger-aware masking;
- hyperparameter tuning;
- curriculum changes;
- opponent training;
- neural networks;
- reuse, migration or conversion of Task 1 or Task 2 Q-tables;
- comparison of parent-prior initialization strategies.

## Definition of done

- [ ] The state-representation mode is configurable and persisted.
- [ ] Resuming with an incompatible state representation is rejected.
- [ ] Every compact feature has unit tests.
- [ ] Control and candidate run plans pass dry-run validation.
- [ ] The experiment is approved before training.
- [ ] Results, figures and the registered decision are documented.
- [ ] AI assistance is documented.
